### PR TITLE
Do not display product in breadcrumb if the page type is 404

### LINKED ISSUE #1673: [2.3.4] Breadcrumb bei Fehlerseite wenn Produkt auf nicht veröffentlichter Seite
Wenn ein Produkt aufgerufen wird, das sich auf einer nicht veröffentlichten Seite befindet, wird korrekt ein 404-Fehler erzeugt.

Jedoch wird das Produkt auf der Fehlerseite in der Breadcrumb-Navigation angezeigt, dies dürfte jedoch nicht sein, denn nicht gefundene Elemente werden in der Breadcrumb-Navigation grundsätzlich nicht angezeigt.

### PR DESCRIPTION
**Pull request by qzminski**
Fixes #1673

### CODE REVIEW

**Comment by aschempp** [comment]
Please rewrite to use early exists. 

```php
if (if ($objPage->type === 'error_404')) {
    return;
}
```

Also apply to the other existing if-loops please.

File: system/modules/isotope/library/Isotope/Frontend.php [hunk 1]
```diff
@@ -630,36 +630,39 @@ public static function getPagesInCurrentRoot(array $arrPages, $objMember = null)
...
+        if (!($alias = Input::getAutoItem('product', false, true))
+            || ($objProduct = Product::findAvailableByIdOrAlias($alias)) === null
+        ) {
+            return $arrItems;
+        }
+
+        /** @var \PageModel $objPage */
+        global $objPage;
+
+        if ($objPage->type === 'error_404') {
```
Comment (aschempp) [inline system/modules/isotope/library/Isotope/Frontend.php]: we should do this check before the *expensive* search for product in the database.

Comment (qzminski) [inline system/modules/isotope/library/Isotope/Frontend.php]: Good point, just changed.

**Comment by aschempp** [comment]
Should we do the same for 403 page?

**Comment by qzminski** [comment]
I don't think so.

**Comment by baumisimon** [comment]
Now the Product never shows in the breadcrumb:
line 636 of Frontend.php
```
if (!($objPage->type instanceof \PageRegular)
```
always returns true ($objPage->type isn't an instance of PageRegular).
If I replace this line with
```
if (!($objPage->type == 'regular')
```
it works as expected for me.

**Comment by qzminski** [comment]
True, it should be

```
if (!($objPage instanceof \PageRegular)
```

**Comment by aschempp** [comment]
Fixed in f13e7a41e94b6ee909113b4dd007fbb497434c94

Next time please open a new issue if something is closed and released so it can be correctly assigned to a milestone.

**Comment by aschempp** [comment]
> True, it should be
> 
> ```
> if (!($objPage instanceof \PageRegular)
> ```

Fun fact: `$objPage` is a `PageModel`, never `PageRegular` in my setup 🤷‍♂

**Pull request merged**